libraries/AP_InertialSensor/AP_InertialSensor_MPU6000.cpp

```diff
@@ -388,7 +388,10 @@ void AP_InertialSensor_MPU6000::start()
     hal.scheduler->resume_timer_procs();
 
     // start the timer process to read samples
-    _dev->register_periodic_callback(1000, FUNCTOR_BIND_MEMBER(&AP_InertialSensor_MPU6000::_poll_data, bool));
+    if (_dev->register_periodic_callback(1000, FUNCTOR_BIND_MEMBER(&AP_InertialSensor_MPU6000::_poll_data, bool)) == nullptr) {
```

Comment:
I still don't think this is correct. You will actually never enter this branch of the code because you are checking the return code of the Device implementation that has no idea on BusDriver/AuxiliaryDevice.

Calling on a periodic callback or on a timer thread doesn't change anything either.

Correct fix IMO would be to continue registering the periodic callback and let driver using it (HMC5883 / AK8963) to read the device in its own pace, like it's doing now.